lib/FaceLandmarksExtractor/FaceLandmarksExtractor.py

```diff
@@ -143,12 +143,11 @@ def extract(input_image, detector, verbose, all_faces=True, scale_to=2048):
     (h, w, ch) = input_image.shape
 
     input_scale = scale_to / (w if w > h else h)
-    input_image = cv2.resize (input_image, ( int(w*input_scale), int(h*input_scale) ), interpolation=cv2.INTER_LINEAR)
-    input_image_bgr = input_image[:,:,::-1].copy() #cv2 and numpy inputs differs in rgb-bgr order, this affects chance of dlib face detection
-    input_images = [input_image, input_image_bgr]
+    input_image = cv2.resize (input_image, ( int(w*input_scale), int(h*input_scale) ), interpolation=cv2.INTER_LANCZOS4)
+    input_image = input_image[:,:,::-1] #cv2 loads into BGR format, dlib model is trained on RGB format, switch to RGB
  
     detected_faces = []
-    for current_detector, input_image in ((current_detector, input_image) for current_detector in dlib_detectors for input_image in input_images):
```

Comment:
lmao, the chaining of these for loops has lost me. Why do you remove them here?

Reply:
it's pretty odd to me too. I understand it to mean that if there are multiple detectors ( i.e. hog and CNN ) and multiple images in input_images( BGR and RGB), loop over every unique combination of the two and send that combo to the current_detector call. the following len break line aborts out of this loop on the first image with found faces and it won't check any remaining detector/image combos. ( if you load the hog model first, this would never check CNN model.... , if you load BGR first, this would never check RGB ) . As Dlib firmly states his model is trained on RGB faces, you want that image first in the list ( now second and mislabeled as bgr). on limited testing, found no difference in number of found faces when eliminating the list of two images and just using only the RGB. testing on several thousand extracts today, so should be able to report back if any.

Reply:
This is the part that was running the RGB and BGR though. It was making a list of all detectors, and then the image in RGB and BGR then looping through them each so that every detector was fed both RGB and BGR. That is the whole reason for the PR.

The change is to replace the double loops (one loop on dlib_detectors and one loop on input_images = [input_image, input_image_bgr]). By removing the BGR version, he didn't need to loop on both, so only loops on the dlib_detectors so all detectors are used on the RGB version of the image only.

Reply:
Have you guys seen this #236 . Apparently recognition was not good without bgr

Reply:
#236 is a not exactly proof that BGR helps, since it was only BGR going to the detector before that patch and patch added RGB. The patch obviously improved the results, but the question now is if keeping the BGR is necessary. This patch removes the BGR and just send the RGB to the detector, so it needs some testing, but shows promise at this time.